recipes/ipaddress/all/conanfile.py

```diff
@@ -0,0 +1,72 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import copy, get
+from conan.tools.layout import basic_layout
+from conan.tools.scm import Version
+from conan.tools.microsoft import is_msvc
+import os
+
+
+required_conan_version = ">=1.52.0"
+
+
+class IpAddressConan(ConanFile):
+    name = "ipaddress"
+    description = "A library for working and manipulating IPv4/IPv6 addresses and networks"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/VladimirShaleev/ipaddress"
+    topics = ("ipv4", "ipv6", "ipaddress", "ip", "network", "header-only")
+    package_type = "header-library"
+    settings = "os", "arch", "compiler", "build_type"
+    no_copy_source = True
+
+    options = {"exceptions": [True, False], "overload_std": [True, False], "ipv6_scope": [True, False], "ipv6_scope_max_length": ["ANY"]}
+    default_options = {"exceptions": True, "overload_std": True, "ipv6_scope": True, "ipv6_scope_max_length": 16}
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def package_id(self):
+        self.info.clear()
+
+    def validate(self):
+        compiler = self.settings.compiler
+        compiler_version = Version(self.settings.compiler.version)
+
+        if compiler == "gcc" and compiler_version < "7.5":
+            raise ConanInvalidConfiguration(f"{self.ref} doesn't support gcc < 7.5")
+        if compiler == "clang" and compiler_version < "6.0":
+            raise ConanInvalidConfiguration(f"{self.ref} doesn't support clang < 6.0")
+        if compiler == "apple-clang" and compiler_version < "13.0":
+            raise ConanInvalidConfiguration(f"{self.ref} doesn't support Apple clang < 13.0")
+        if is_msvc(self) and compiler_version < "16":
+            raise ConanInvalidConfiguration(f"{self.ref} doesn't support MSVC < 16")
```

Comment:
Not wrong, but it would be better to follow the conventional format:
https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/header_only/all/conanfile.py#L30-L75

Reply:
Yes, it will probably be better. I'll do it as in the template. But let me change the error text to more accurately convey the cause of the error.

If the new text is unacceptable, please respond to me. I will return the text from template.